specs/source.go

```diff
@@ -94,6 +100,10 @@ func (s *Source) Validate() error {
 		}
 	}
 
+	if s.Scheduler != "dfs" && s.Scheduler != "round-robin" {
```

Comment:
nit: if you make scheduler option a typed one, there's a quick check pattern:
```go
type SchedulerFlavor byte
const (
    SchedulerDFS SchedulerFlavor = iota
    SchedulerRR
    _badSchedulerOption
)

...
if s.Scheduler >= _badSchedulerOption { ...
```

Reply:
An import cycle prevents us from using the constants from the other package here. We'd have to put the constants somewhere else first, maybe to this package

Reply:
Updated by moving the constants to this package